src/validator/CreditCardValidator.java

```diff
@@ -0,0 +1,62 @@
+package validator;
+
+import java.util.ArrayList;
+
+public class CreditCardValidator implements CreditCardChecking {
+
+	@Override
+	public boolean checkValidity(CreditCardVendor ccv, String number) {
+		ArrayList<Integer> chars = ccv.getNumberOfChars();
+		ArrayList<String> masks = ccv.getIIN();
+		
+		boolean result1 = checkLengthCorrectness(number, chars);
+		boolean result2 = checkIINMaskCorrectness (number, masks);
+		boolean result3 = checkLuhnalgorithmCorrectness(number);
```

Comment:
Another issue is that currently, it would be hard to test private methods with unit tests. I think that methods: checkLengthCorrectness, checkIINMaskCorrectness, and checkLuhnalgorithmCorrectness should be separated to separate classes. Single responsibility principle would not be broken. 

Additionally, we could create a common interface for these 3 CorrectnessCheckers. In the future, when another checker would be added, it will be enough to implement an interface and a check method. 
Our class would meet the Open–closed principle: "Software entities ... should be open for extension, but closed for modification."

Reply:
Additionally, we could create a common interface for these 3 CorrectnessCheckers. In the future, when another checker would be added, it will be enough to implement an interface and a check method.
Our class would meet the Open–closed principle: "Software entities ... should be open for extension, but closed for modification." New extensions would be simple.